client/main.lua

```diff
@@ -521,35 +521,44 @@ function OpenFineMenu(player)
 	end)
 end
 
+local fineList = {}
 function OpenFineCategoryMenu(player, category)
-	ESX.TriggerServerCallback('esx_policejob:getFineList', function(fines)
-		local elements = {
-			{unselectable = true, icon = "fas fa-scroll", title = TranslateCap('fine')}
+	if not fineList[category] then
+		ESX.TriggerServerCallback('esx_policejob:getFineList', function(fines)
+			fineList[category] = fines
+		end, category)
+
+		while fineList[category] == nil do
+			Wait(50)
+		end	
+	end
```

Comment:
if possible do not use a while loop

```suggestion
if not fineList[category] then
		local p = promise.new()

		ESX.TriggerServerCallback('esx_policejob:getFineList', function(fines)
			p:resolve(fines)
		end, category)

		fineList[category] = Citizen.Await(p)
end
```